### PR TITLE
Inc 21/profile settings

### PR DESCRIPTION
**Pull request by irinrudko**
1. add profileApi (get and add user profile data). add userRoutes paths for it.
2. create userSlice to store user data.
3. create tabs using radix tabs and separate Tab component. add url when switching tabs.
4. create InputForm components to use in forms.
5. create general information component which gets data from store . 
6. install redux-persist lib (use pnpm to get it) to save redux data when refreshing page (see changes in store and _app.tsx )
7. add useAppDispatch and useAppSelector hooks (according to RTK docs) not to duplicate types every time. Please use them instead of useSelector/useDispatch

TODO:
– date picker (INC-19)
– refactor (code style and change some names)
– input validation
– fix types in General info component/profileApi
– translation
– fix styles
– folder structure (common components, api, slices)

image

### CODE REVIEW

File: pages/home/index.tsx
Comment (irinrudko) [inline pages/home/index.tsx:1]: кнопка-заглушка. в ТЗ мы должны попасть на general info по тыку на кнопку

**Review by Viarcinski — APPROVED** [review]
Looks good

**Review by maskpa100 — COMMENTED** [review]
ok

**Review by ilich01 — APPROVED** [review]

**Review by maskpa100 — APPROVED** [review]

File: components/profile/Tabs/ProfileTabs.module.scss [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  flex: 1 0 auto;
+  border-radius: 4px;
+  padding: 0 10px;
+  font-size: 15px;
+  line-height: 1;
+  color: var(--violet-11);
+  box-shadow: 0 0 0 1px var(--violet-7);
+  height: 35px;
+}
+.Input:focus {
```
Comment (TURP1) [inline components/profile/Tabs/ProfileTabs.module.scss:77]: &:focus

File: components/profile/Tabs/ProfileTabs.module.scss [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  align-items: center;
+  justify-content: center;
+  border-radius: 4px;
+  padding: 0 15px;
+  font-size: 15px;
+  line-height: 1;
+  font-weight: 500;
+  height: 35px;
+}
+.Button.green {
```
Comment (TURP1) [inline components/profile/Tabs/ProfileTabs.module.scss:92]: use &

File: components/profile/Tabs/ProfileTabs.module.scss [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  background-color: var(--green-4);
+  color: var(--green-11);
+}
+.Button.green:hover {
```
Comment (TURP1) [inline components/profile/Tabs/ProfileTabs.module.scss:96]: &:hover

File: package.json [hunk 4]
```diff
@@ -25,11 +29,14 @@
     "react-i18next": "^13.0.1",
     "react-redux": "^8.1.1",
     "react-toastify": "^9.1.3",
-    "typescript": "5.1.3"
+    "redux-persist": "^6.0.0",
```
Comment (TURP1) [inline package.json:32]: do we need persist?

File: package.json [hunk 5]
```diff
@@ -11,10 +11,14 @@
   },
   "dependencies": {
     "@radix-ui/react-form": "^0.0.3",
+    "@radix-ui/react-tabs": "^1.0.4",
     "@reduxjs/toolkit": "^1.9.5",
+    "@types/axios": "^0.14.0",
```
Comment (TURP1) [inline package.json:16]: to dev.dependencies

**Pull request closed without merging**